SEED-platform/seed
# 2335 refactor/rename column list profiles

#### What's this PR do? 
Renames the ColumnListSettings model to ColumnListProfile. Updates terminology on the frontend and backend to accommodate this change.

Note: to maintain v2 and v2.1 API's prior to sunsetting, the column_list_setting view was not changed for those endpoints. 

In addition to the rename of the model, there was a need to change a model field from `settings_location` to `profile_location`.  This change was proliferated throughout the codebase to reflect the model name refactor including `api/v3/column_mapping_profiles`

#### How should this be manually tested?
Apply the migrations to update the schemas. Hard refresh your browser page to reset the cache

Column List profiles are used in a couple of places:

- Inventory List View -> (Prop/Taxlot List tab + Cross-Cycles tab) -> both (Properties and Taxlots)
  - Verify "Column List Profiles" are populated in the dropdown menu.

- Inventory List View -> List Settings -> (both Properties and Taxlots)
  - Verify "Column List Profiles" are populated in the dropdown menu.
  - try: creating, updating, changing the name, and deleting profiles

- Inventory Detail View -> (Prop/Taxlot Detail tab + Cross-Cycles tab) -> (both Properties and Taxlots)
  - Verify "Detail Column List Profiles" are populated in the dropdown menu.

- Inventory Detail View -> Settings -> (both Properties and Taxlots)
  - Verify "Detail Column List Profiles" are populated in the dropdown menu.
  - try: creating, updating, changing the name, and deleting profiles
  

#### What are the relevant tickets?
#2335
